### PR TITLE
Page asset path

### PR DESCRIPTION
**Pull request by oturpe**
Greetings, 

I added a new tag that allows referring to page's asset path from contexts other than page itself, referring to it by page id. I use this on http://koti.kapsi.fi/oturpe/projects/ to fetch cover image of each page for page listing. Do you think this would be a useful complement to _asset_path_ tag?

### CODE REVIEW

**Comment by samrayner** [comment]
Hey @oturpe, thanks for the PR.

I think it might be nicer to either:

a) Make the post ID an additional, optional argument on the same tag, i.e. `{% asset_path document.pdf /2012/05/25/another-post-title %}`.

or b) just release this yourself as a separate Jekyll tag

WDYT?

**Comment by oturpe** [comment]
Hi Sam!

I agree that having two separate tags is somewhat complex solution for the small amount of functionality we are discussing here. The reason for such choice was to avoid changing the semantics of `asset_path` itself in any way. But in case you do not think it matters if there is some small change, I can go with your suggestion A. I suppose the main difference would be that

```
{% asset_path filename with space.ext %} 
```
will not resolve as it used to - instead it would have to be
```
{% asset_path "filename with space.ext" %}
```

So: first quoted string is the asset file name. If there are no quotes it is the part preceding the first whitespace. If non-whitespace string follows, it is the post id.

Does this sound ok to you?

Otto

**Comment by oturpe** [comment]
That is another good way of doing it. I think we have a mutual understanding how it should work. I will work on it when I have time (which might take some time) and get back to you when I get something ready.

**Comment by oturpe** [comment]
Hello again,

I finally found some time to look into this. See commit a4e41cb comment and contents to see what was done. What do you think?

File: README.md [hunk 1]
```diff
@@ -48,5 +60,29 @@ then the tag will output:
...
 ```
 
+When used with `post_id` parameter, the tag is useful for showing asset from each page. Given the site contains pages:
+```
+post-title
+another-post-title
+```
+then
+```
+{% for post in site.posts %}{% asset_path cover.jpg {{post.id}} %}{% endfor %} on index.html
```
Comment (samrayner) [inline README.md]: `on index.html` should be outside code block

File: README.md [hunk 2]
```diff
@@ -5,7 +5,11 @@ Jekyll Asset Path Tag
 
 A liquid tag to output a relative URL for assets based on the post or page, allowing you to organise your assets into subdirectories.
 
-Syntax: `{% asset_path [filename] %}`
+Syntax:
+```
+{% asset_path filename [post_id] %}
```
Comment (samrayner) [inline README.md]: Best to remove the square brackets to avoid confusion.

File: README.md [hunk 3]
```diff
@@ -5,7 +5,11 @@ Jekyll Asset Path Tag
...
+{% asset_path "filename with whitespace" [post_id] %}
```
Comment (samrayner) [inline README.md]: Best to remove the square brackets to avoid confusion.

File: asset_path_tag.rb [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
-# Syntax {% asset_path [filename] %}
+# Syntax
+#    {% asset_path filename [post_id] %}
+#    {% asset_path "filename with whitespace" [post_id] %}
 #
 # Examples:
 # {% asset_path kitten.png %} on post 2013-01-01-post-title
 # {% asset_path pirate.mov %} on page page-title
+# {% asset_path document.pdf /2012/05/25/another-post-title %}
+# {% asset_path "document with long name.pdf" /2012/05/25/another-post-title %}
```
Comment (samrayner) [inline asset_path_tag.rb]: `# {% asset_path "document with spaces in name.pdf" /2012/05/25/another-post-title %}`

File: asset_path_tag.rb [hunk 5]
```diff
@@ -1,16 +1,25 @@
...
 #
 # Output:
 # /assets/posts/post-title/kitten.png
 # /assets/page-title/pirate.mov
+# /assets/posts/another-post-title/document.pdf
+# /assets/posts/another-post-title/document with long name.pdf
```
Comment (samrayner) [inline asset_path_tag.rb]: `# /assets/posts/another-post-title/document with spaces in name.pdf`

File: asset_path_tag.rb [hunk 6]
```diff
@@ -38,36 +80,30 @@ def initialize(tag_name, markup, tokens)
 
     def render(context)
       if @markup.empty?
-        return "Error processing input, expected syntax: {% asset_path [filename] %}"
+        return "Error processing input, expected syntax: {% asset_path filename [post id] %}"
```
Comment (samrayner) [inline asset_path_tag.rb]: Remove square brackets

**Comment by oturpe** [comment]
Here you go

**Comment by samrayner** [comment]
Thanks @oturpe!

**Pull request merged**